### ISSUE TITLE
Disposing component causing refetches

### ISSUE THREAD

**Issue by bjtrounson** [Jcparkyn/Phetch#6]
Hi before I get into my issue just want to thank you a lot for this library it has made my transition from doing React development in my previous job to using Blazor quite smooth.

## Problem
I have a parent component with the same ObserveQuery as a nested child. The nested child is within a tab menu and gets disposed of if the user switches to a different menu. When it loads back in it recalls the endpoints refetching the data which I would prefer it not to and pull from the current state of the parent as the amount of data it's refetching is of decent size. I know this could be fixed with prop drilling the data or context from the parent ObserveQuery, but I feel like it's going to cause horrible DX as my app grows in size and I have multiple components using this query / state.

This is pseudo-ish code to visually explain what I'm talking about and kinda what I'm doing in my app.
### Parent
```csharp
<ObserveQuery Target="@_exampleQuery" OnChanged="StateHasChanged"/>
<div>
    @if(menuId == "example") {
         <ChildComponent />
     } else {
         <span>testing</span>
     }
</div>
@code {
     private Query<Example[]?> _exampleQuery = null!;
     
     protected override  async Task OnInitializedAsync()
     {
	    _exampleQuery = ExampleEnpoints.Example.Use();
     }
}
```

### Child
```csharp
<ObserveQuery Target="@_exampleQuery" OnChanged="StateHasChanged"/>
<div>
    <span>exampleQuery.Data</span>
</div>

@code {
     private Query<Example[]?> _exampleQuery = null!;
     
     protected override  async Task OnInitializedAsync()
     {
	    _exampleQuery = ExampleEnpoints.Example.Use();
     }
}
```

### Questions
I'm not fully sure how the caching works so this in my uneducated understanding, I know the ObserveQueries are at least linked together since when the child refetches it will trigger the loading state on the parent. I'm assuming this is through the ExampleEndpoint since it is loaded as a scoped service which is basically a singleton in Blazor WASM from my understanding. Would the cache not simply carry over if these Endpoints are somewhat singletons, or does it get purged when a new ObserveQuery subscribes to the endpoint? 

I might be trying to use this library way out of it's use case so let me know if this is the case and if you have any suggestion of a different library that might get me something similar, but I do really like this one and would prefer to use it.

**Comment by Jcparkyn** [comment]
I'm glad to hear you're liking the library.

If I'm understanding your problem correctly, this is the intended default behavior, but you can very easily override it by setting a `DefaultStaleTime` in the endpoint options. You can also set the `StaleTime` for individual queries if you want it to be different per usage. The readme could probably use some updating here, but most of the details are in the [XML docs](https://www.fuget.org/packages/Phetch.Core/0.5.1/lib/netstandard2.1/Phetch.Core.dll/Phetch.Core/QueryOptions) for the individual options.

From your description it sounds like you might also be `IsFetching` when `IsLoading` might fit your use case better? `IsLoading` should stay false in the parent component while the child re-fetches*, so it's much less intrusive to the user if you ever expect to be re-fetching in the background.

*If that's not what's happening, let me know because that's probably a bug.

**Comment by bjtrounson** [comment]
Thanks for the quick reply.

That seems to work great and yes I was using IsFetching I've switched it to IsLoading. Another question if the StaleTime is set to for example TimeSpan.MaxValue will the Query still fetch data if there is no cache from a parent or previous query?

Also if you are needed help with documentation write or any dev work I wouldn't mind giving a hand.

Really appreciate the help :)

**Comment by Jcparkyn** [comment]
> if the StaleTime is set to for example TimeSpan.MaxValue will the Query still fetch data if there is no cache from a parent or previous query?

Yes, queries will always be fetched if there is no cached data, regardless of the `StaleTime`. This includes if a query is used in two places with different arguments: both will be fetched. Another thing to keep in mind is the `CacheTime`, which will automatically remove queries from the cache if they are unused for a long enough time.

Re. contributing, the most useful things would be:
1. Just use the library normally and report any issues you run into. It's pretty much feature-complete (aside from some nice-to-haves), so the only real reason I haven't published a v1.0 yet is that it hasn't had enough usage and testing to uncover all the issues.
2. If you want to contribute directly, there are quite a few more unit tests needed, particularly for the Blazor components. You can look at the existing tests for an example of how to write these.
3. Documentation is ironically one of the harder things to contribute to, but a short section in the readme like "CacheTime and StaleTime" (after "Invoking queries manually") with the info from this thread might be useful. Also feel free to add any details that are missing in the XML comments.
